### PR TITLE
feat: add a script to help user generating a token and a new password

### PR DESCRIPTION
**Pull request by ykatchou**
Small PR to improve user UX while handling password.

### CODE REVIEW

File: bin/user_helper.py [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+import os
+import typer
+
+# To run this script, 
+# You need to be in avatar-python repository
+# Then poetry run ./bin/user_helper.py
+
+app = typer.Typer(help="""
+This script allow to generate a token and then change your password.
+You are expected to call generate_token first, then change_password.
```
Comment (tcrasset) [inline bin/user_helper.py:12]:
```suggestion
This script allows to generate a token and then change your password.
You are expected to call generate_token first, then change_password.
```

File: bin/user_helper.py [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+""")
+
+def load_client(base_url: str) -> ApiClient:
```
Comment (tcrasset) [inline bin/user_helper.py:15]:
```suggestion
def load_client(base_url: Optional[str]) -> Optional[ApiClient]:
```

File: bin/user_helper.py [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+        print("ERROR: Base url should be provided, you can do it through the env variable AVATAR_BASE_URL.")
+        return None
+    return ApiClient(base_url=base_url)
+
+@app.command()
+def generate_token(email: str, base_url:str=os.environ.get("AVATAR_BASE_URL"), ):
+    """
+    Generate a validation token which is done by providing the application url and only an email
+    """
+    client = load_client(base_url)
```
Comment (tcrasset) [inline bin/user_helper.py:26]:
```suggestion
    client = load_client(base_url)
    if not  client:
        return
```

File: bin/user_helper.py [hunk 4]
```diff
@@ -0,0 +1,44 @@
...
+    print(f"Resetting password for {email}, check your email for your token")
+    client.forgotten_password(email)
+
+@app.command()
+def change_password(email: str, token: str, new_password: str, new_password_confirm: str, base_url:str=os.environ.get("AVATAR_BASE_URL"), ):
+    """
+    Resetting the account with a password, which is done by providing all parameters.
+    """
+    client = load_client(base_url)
```
Comment (tcrasset) [inline bin/user_helper.py:35]: if....

**Review by tcrasset — COMMENTED** [review]
Is this for our use or for the user? I think most people won't clone the repo, so this might not get used as much as we think

**Review by tas17 — COMMENTED** [review]
+1 for using typer :)
However, I'm not sure who would use this ?
- Most users won't have access to `bin` dir with our release system
- Our doc is pretty clear on this (https://python.docs.octopize.io/latest/tutorial.html#reset-your-password), and both snipets are only 2 lines of code, so does not look very helpful, even for us ?

**Comment by ykatchou** [comment]
My idea is, it’s not a nice UX for user to start on our app to need to white python script directly.
So I wanted to make it improved at first.
Do you have a better idea? maybe a dedicated "webpage" just for that password?

**Comment by ykatchou** [comment]
As suggested by Tom, a notebook might be a better compromise

**Comment by ykatchou** [comment]
Aborted.

**Pull request closed without merging**